cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,513 @@
+    # the pattern enables searching for the same.
+
+    logs_pattern = re.compile(rf'{re.escape(logs_pattern)}/.+')
+    # Find all strings in the array that match the pattern
+    gcs_log_objects = [
+        gcs_log_object
+        for gcs_log_object in gcs_bucket_contents
+        if logs_pattern.match(gcs_log_object)
+    ]
+    for gcs_log_object in gcs_log_objects:
+        # -1 is returned in case metric not found in the log file.
+        (metric_value, is_query_result_present) = get_blob_and_check_metric(
+            gcs_log_object,
+            cluster_temp_bucket,
+            metric_string,
+            end_of_metric_string,
+            query,
+        )
+        if metric_value != -1:
+            is_metric_found = True
+            # Sum up all the values.
+            total_metric_count += metric_value
+
+        # Check if query result is found in any one of the files.
+        # True if found in at least one of the files.
+        # False in case not found in any.
+        if is_query_result_present:
+            is_query_result_found = True
+
+    # If query has been set, check if query results were obtained any
+    # at least one of the logs. If not raise an Exception.
+    if query and not is_query_result_found:
+        raise RuntimeError(
+            '[Log: parse_logs ERROR] Unable to find the query results in any of the'
+            ' logs'
+        )
+    # If found in any of the logs, return the value, else raise an error.
+    if is_metric_found:
+        return total_metric_count
+    raise RuntimeError(
+        f'[Log: parse_logs ERROR] Unable to find the "{metric_string}" in any of'
+        ' the logs'
+    )
+
+
+def run(
+    cluster_project_name,
+    cluster_name,
+    region,
+    job_id,
+    arg_project,
+    arg_dataset,
+    arg_table,
+    query,
+):
+    """Method that calls all the helper function to determine success of a job.
+
+    Args:
+      cluster_project_name: Project ID of the GCP project that contains the
+        cluster.
+      cluster_name: name of the cluster on which the dataproc job is running.
+      region: region in which the cluster runs.
+      job_id: JOB_Id of the dataproc job.
+      arg_project: Resource project id (from which rows are read)
+      arg_dataset: Resource dataset name (from which rows are read)
+      arg_table: Resource table name (from which rows are read)
+      query: String containing the query incase needs to be run by the connector.
+
+    Raises:
+      AssertionError: When the rows read by connector and in the BQ table do not
+      match.
+    """
+    # 1. Get the temp bucket name.
+    cluster_temp_bucket = get_cluster_temp_bucket(
+        cluster_project_name, cluster_name, region
+    )
+    # 2. get the pattern of logs inside the temp bucket in GCS.
+    logs_pattern = get_logs_pattern(cluster_project_name, region, job_id)
+    # Read the blob and get the metric from them
+    metric = read_logs(cluster_temp_bucket, logs_pattern, query)
```

Comment:
We should either number all steps or none. I'd suggest removing these numbers. Also, either all comments follow proper punctuation (like full stops) or none. Let's be consistent.

Reply:
The consistency in comments is required for all comments in this PR. Let's be conscious of this.

Reply:
Fixed.